### PR TITLE
Dependency report

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

1. Finds all containers with github/ministryofjustice `url`s
2. Clones them
3. Reads the circleci orb version from `.circleci/config.yml`
4. Reads the gradle-sprint-boot version from `build.gradle.kts`
5. Generates `exports/docs/dependencies.md`
6. Will publish it to https://structurizr.com/share/56937/documentation

I removed the `git pull` functionality as it was terribly slow. CI will pull fresh anyway

Generated file: https://gist.github.com/sldblog/bb171733565523e0469c62dca3772ddb:
image

## What is the intent behind these changes?

Make collecting and updating shared library data easier than spreadsheets :)

### CODE REVIEW

**Comment by sldblog** [comment]
cc @swilson-mojhub

File: src/main/kotlin/model/Delius.kt [hunk 1]
```diff
@@ -63,7 +63,7 @@ class Delius private constructor() {
         "Delius API",
         "API over the nDelius DB used by HMPPS Digital team applications and services", "Kotlin"
       ).apply {
-        url = "https://github.com/ministryofjustice/delius-api"
+        url = "https://github.com/ministryofjustice/hmpps-delius-api"
```
Comment (mxco86) [inline src/main/kotlin/model/Delius.kt:66]: Thanks!

File: src/main/kotlin/defineDocumentation.kt [hunk 2]
```diff
@@ -14,26 +12,68 @@ fun defineDocumentation(workspace: Workspace) {
...
   w.println("| Software System | API name | Purpose | Links |")
   w.println("| --- | --- | --- | --- |")
 
-  containersWithAPIDocs(model)
+  containersWithAPIDocs(containersWithGit)
     .also { println("[defineDocumentation] Found ${it.size} APIs with documentation URLs") }
     .forEach { writeAPI(w, it) }
 }
 
+private fun writeDependencies(w: PrintWriter, containersWithGit: List<Container>) {
```
Comment (mxco86) [inline src/main/kotlin/defineDocumentation.kt:43]: Might be worth looking at a Markdown generation library for these docs?

**Review by mxco86 — APPROVED** [review]
👍

File: src/main/kotlin/defineDocumentation.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/defineDocumentation.kt:43]: Definitely!

**Pull request merged**